UnitTests/UnitTests.UWP/UI/Controls/Test_Carousel.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.UI.Xaml.Automation;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Markup;
+using Microsoft.Toolkit.Uwp.UI.Automation.Peers;
+using Microsoft.Toolkit.Uwp.Extensions;
+using Carousel = Microsoft.Toolkit.Uwp.UI.Controls.Carousel;
+
+namespace UnitTests.UWP.UI.Controls
+{
+    [TestClass]
+    [TestCategory("Test_Carousel")]
+    public class Test_Carousel : VisualUITestBase
+    {
+        [TestMethod]
+        public async Task ShouldConfigureCarouselAutomationPeerAsync()
+        {
+            await App.DispatcherQueue.EnqueueAsync(async () =>
+            {
+                const int selectedIndex = 1;
+                var items = new ObservableCollection<PhotoDataItem> { new PhotoDataItem { Title = "Hello" }, new PhotoDataItem { Title = "World" } };
+                var carousel = new Carousel { ItemsSource = items };
+
+                await SetTestContentAsync(carousel);
+
+                // Sets the selected item to "World" from the XAML above.
+                carousel.SelectedIndex = selectedIndex;
+
+                const string automationName = "MyAutomationPhotoItems";
+                const string name = "MyPhotoItems";
+
+                var carouselAutomationPeer =
+                    FrameworkElementAutomationPeer.CreatePeerForElement(carousel) as CarouselAutomationPeer;
+
+                Assert.IsNotNull(carouselAutomationPeer, "Verify that the AutomationPeer is CarouselAutomationPeer.");
+                Assert.IsFalse(carouselAutomationPeer.CanSelectMultiple, "Verify that CarouselAutomationPeer.CanSelectMultiple is false.");
+                Assert.IsTrue(carouselAutomationPeer.IsSelectionRequired, "Verify that CarouselAutomationPeer.IsSelectionRequired is true.");
+
+                carousel.SetValue(AutomationProperties.NameProperty, automationName);
+                Assert.IsTrue(carouselAutomationPeer.GetName().Contains(automationName), "Verify that the UIA name contains the given AutomationProperties.Name of the Carousel.");
+
+                carousel.Name = name;
+                Assert.IsTrue(carouselAutomationPeer.GetName().Contains(name), "Verify that the UIA name contains the given Name of the Carousel.");
```

Comment:
Proposal: Don't use dedicated variables for UIAName and name and instead write to property and read that out for mor concise code.

Reply:
So I do this, and similarly with your comment around the selected index constant so that I clearly separate out what my expectations are.

Reply:
While it might make the code more concise, as a test, it feels like it gets harder to read. 

Happy to take a view on this and make the change mind.

Reply:
Hmm I see. My expectation would be that code matches the actual property on the control, not some variable (to put it exaggerated). My question when reading that kind of code would be: "Is the variable different to the actual selected index of the control? Is the behavior not actually relying on the property?"
But I guess it comes down to personal preferences.